lib/galaxy/util/permutations.py

```diff
@@ -16,13 +17,16 @@
     MULTIPLIED="multiplied",
 )
 
+# generic type of splitting input dictionary
+T = TypeVar("T")
+
 
 class InputMatchedException(MessageException):
     """Indicates problem matching inputs while building up inputs
     permutations."""
 
 
-def expand_multi_inputs(inputs, classifier, key_filter=None):
+def expand_multi_inputs(inputs: Dict[str, T], classifier, key_filter=None):
```

Comment:
I think here and below we can just use `Dict[str, Any]` without generics?

Reply:
We could but we're returning dictionaries of the same type so when we fill out the output declarations of these functions we will get more precise typing with the generics.